Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/DriverOnboardingV2.hs

```diff
@@ -328,3 +334,58 @@ postDriverRegisterSsn (mbPersonId, _, _) API.Types.UI.DriverOnboardingV2.SSNReq
           driverId = driverId',
           ssn = ssn'
         }
+
+postDriverRegisterPancard ::
+  ( ( Kernel.Prelude.Maybe (Kernel.Types.Id.Id Domain.Types.Person.Person),
+      Kernel.Types.Id.Id Domain.Types.Merchant.Merchant,
+      Kernel.Types.Id.Id Domain.Types.Merchant.MerchantOperatingCity.MerchantOperatingCity
+    ) ->
+    API.Types.UI.DriverOnboardingV2.DriverPanReq ->
+    Environment.Flow Kernel.Types.APISuccess.APISuccess
+  )
+postDriverRegisterPancard (mbPersonId, merchantId, _) req = do
+  personId <- mbPersonId & fromMaybeM (PersonNotFound "No person found")
+  person <- runInReplica $ PersonQuery.findById personId >>= fromMaybeM (PersonNotFound personId.getId)
+  mbPanInfo <- SQDPC.findByPanNumber req.panNumber
+  getImage req.imageId1 personId ------- Just checking whether the image exists or not
+  case mbPanInfo of
+    Just pan -> unless (pan.driverId == personId) $ throwImageError req.imageId1 PanAlreadyLinked
+    Nothing -> do
+      panCard <- buildPanCard merchantId person req
+      QDPC.create panCard
+  return Success
+  where
+    getImage :: Kernel.Types.Id.Id Image.Image -> Kernel.Types.Id.Id Domain.Types.Person.Person -> Environment.Flow ()
+    getImage imageId personId = do
+      imageMetadata <- ImageQuery.findById imageId >>= fromMaybeM (ImageNotFound imageId.getId)
+      unless (imageMetadata.isValid) $ throwError (ImageNotValid imageId.getId)
+      unless (imageMetadata.personId == personId) $ throwError (ImageNotFound imageId.getId)
+      unless (imageMetadata.imageType == DTO.PanCard) $
+        throwError (ImageInvalidType (show DTO.PanCard) (show imageMetadata.imageType))
+
+buildPanCard ::
+  Kernel.Types.Id.Id Domain.Types.Merchant.Merchant ->
+  Domain.Types.Person.Person ->
+  API.Types.UI.DriverOnboardingV2.DriverPanReq ->
+  Environment.Flow Domain.DriverPanCard
+buildPanCard merchantId person API.Types.UI.DriverOnboardingV2.DriverPanReq {..} = do
+  now <- getCurrentTime
+  id <- generateGUID
+  encryptedPan <- encrypt panNumber
+  return
+    Domain.DriverPanCard
+      { consent = consent,
+        consentTimestamp = now,
+        documentImageId1 = imageId1,
+        documentImageId2 = imageId2,
+        driverDob = Nothing,
+        driverId = person.id,
+        driverName = Just person.firstName,
+        failedRules = [],
+        id = id,
+        panCardNumber = encryptedPan,
+        verificationStatus = Domain.PENDING,
```

Comment:
Why Pending? How will it be verified?

Reply:
So i kept as pending because , once we have mechanism for verification of pan then we can verify it. If you suggest I could change it to verified.